configs/cache_partitioning/cache_partitioning.py

```diff
@@ -0,0 +1,212 @@
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+# This script showcases the functionality of cache partitioning policies,
+# containg a simple system comprised of a memory requestor (TrafficGen),
+# a cache enforcing policies for requests and a SimpleMemory backing store.
+#
+# Using the Way policy, the cache should show the following statistics in the
+# provided configuration:
+#
+# | Allocated Ways | 1 | 2   | 3   | 4   | 5   | 6   | 7   | 8    |
+# |----------------|---|-----|-----|-----|-----|-----|-----|------|
+# | Cache Hits     | 0 | 256 | 384 | 512 | 640 | 768 | 896 | 1024 |
+#
+# Using the MaxCapacity policy, expected results are the following:
+#
+# | Allocation % | 10 | 20  | 30  | 40  | 50  | 60  | 70  | 80  | 90  | 100  |
+# |--------------|----|-----|-----|-----|-----|-----|-----|-----|-----|------|
+# | Cache Hits   | 0  | 152 | 307 | 409 | 512 | 614 | 716 | 819 | 921 | 1024 |
+
+import argparse
+import os
+import sys
+
+import m5
+from m5.objects import *
+from m5.util import addToPath
+
+# gem5 config imorts
+addToPath("../")
+from common.MemConfig import config_mem
+
+SCRIPT_DIR = os.path.dirname(os.path.realpath(__file__))
+
+
+STATIC_OPTIONS = argparse.Namespace(
+    mem_channels=1,
+    mem_type="SimpleMemory",
+    # configure values for SimpleMemory interaface
+    request_latency="10ns",
+    mem_bandwidth="1GiB/s",
+    # simulation settings
+    system_frequency="10THz",
+    max_ticks=2000000000,
+)
+
+
+def verifyPolicyParams(args):
+    """
+    Verify that Requested partitioning policy has been provided with a suitable
+    configuration
+    """
+    if args.policy == "way":
+        assert (
+            type(args.way_allocation) == int
+        ), "Need to supply Way Policy allocation"
+        assert (
+            args.way_allocation >= 0
+        ), "Way Policy needs positive number of ways"
+
+    if args.policy == "max_capacity":
+        assert (
+            type(args.capacity_allocation) == float
+        ), "Need to supply Max Capacity Policy allocation"
+        assert (
+            args.capacity_allocation >= 0
+        ), "Max Capacity Policy needs allocation in range [0, 1]"
+        assert (
+            args.capacity_allocation <= 1
+        ), "Max Capacity Policy needs allocation in range [0, 1]"
+
+
+def generatePartPolicy(args):
+    """
+    Generate Partitioning Policy object based on provided arguments
+    """
+    if args.policy == "way":
+        allocated_ways = []
+        for i in range(0, args.way_allocation):
+            allocated_ways.append(i)
+
+        allocation = WayPolicyAllocation(partition_id=0, ways=allocated_ways)
+        Policy = WayPartitioningPolicy(allocations=[allocation])
+
+    if args.policy == "max_capacity":
+        Policy = MaxCapacityPartitioningPolicy(
+            partition_ids=[0], capacities=[args.capacity_allocation]
+        )
+
+    return Policy
```

Comment:
Policy (which should be all lowercase according to PEP8 I believe) is not defined for other policies (other than way and max-capacity). I believe we should handle that unknown case more gracefully. IMHO we can just return the policy object directly without creating a variable in the other cases

Reply:
I've changed the function to directly return the policy configurations as you suggested. I've also gone ahead and added an assertion for the types of policies it can generate